exonum-java-binding/common/src/main/java/com/exonum/binding/common/proofs/map/UncheckedFlatMapProof.java

```diff
@@ -250,22 +256,35 @@ private CheckedMapProof checkProof() {
   }
 
   private static HashCode getEmptyProofListHash() {
-    return HashCode.fromBytes(new byte[Hashing.DEFAULT_HASH_SIZE_BYTES]);
+    return HASH_FUNCTION.newHasher()
+        .putByte(MAP_NODE_PREFIX)
+        .putBytes(new byte[Hashing.DEFAULT_HASH_SIZE_BYTES])
+        .hash();
   }
 
   private static HashCode getSingleEntryRootHash(MapProofEntry entry) {
-    return getSingleEntryRootHash(entry.getDbKey(), entry.getHash());
+    HashCode entryRootHash = getSingleEntryRootHash(entry.getDbKey(), entry.getHash());
+    return getMapProofRootHash(entryRootHash);
   }
 
   private static HashCode getSingleEntryRootHash(MapEntry<ByteString, ByteString> entry) {
     DbKey dbKey = DbKey.newLeafKey(entry.getKey());
     HashCode valueHash = HASH_FUNCTION.hashByteString(entry.getValue());
-    return getSingleEntryRootHash(dbKey, valueHash);
+    HashCode entryRootHash = getSingleEntryRootHash(dbKey, valueHash);
+    return getMapProofRootHash(entryRootHash);
+  }
+
+  private static HashCode getMapProofRootHash(HashCode merkleRoot) {
+    return HASH_FUNCTION.newHasher()
+        .putByte(MAP_NODE_PREFIX)
+        .putObject(merkleRoot, hashCodeFunnel())
+        .hash();
   }
 
   private static HashCode getSingleEntryRootHash(DbKey key, HashCode valueHash) {
```

Comment:
This patch changes the meaning of the two previous and this method: the former calculate one thing, this one — another. Therefore, they no longer can have the same name.